File: src/main/java/com/summer/lijiahao/utils/OpenNCHomeAction.java
package com.summer.lijiahao.utils;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.ui.Messages;
import com.summer.lijiahao.abs.AbstractAnAction;
import com.summer.lijiahao.base.NccEnvSettingService;

import java.awt.*;
import java.io.File;

public class OpenNCHomeAction extends AbstractAnAction {
    @Override
    public void doAction(AnActionEvent event) {
        try {
            String ncHome = NccEnvSettingService.getInstance().getNcHomePath();
            if (ncHome == null || ncHome.equals("")) {
                throw new Exception("未配置NC home");
            }

            Desktop desktop = Desktop.getDesktop();
            File dirToOpen = new File(ncHome);
            desktop.open(dirToOpen);
        } catch (Exception e) {
            String message = e.getMessage();
            Messages.showInfoMessage(message, "Error");
        }
    }
}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/menu/action/OpenNCHomeAction.kt
package com.summer.lijiahao.nccdevtool.menu.action

import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.ui.Messages
import com.summer.lijiahao.nccdevtool.base.action.AbstractAnAction
import com.summer.lijiahao.nccdevtool.devconfig.service.NCCloudEnvSettingService
import java.awt.Desktop
import java.io.File

class OpenNCHomeAction : AbstractAnAction() {
    override fun doAction(event: AnActionEvent) {
        try {
            val ncHome: String = NCCloudEnvSettingService.getInstance(event).ncHomePath
            if (ncHome.isEmpty()) {
                throw Exception("未配置NC home")
            }
            val desktop = Desktop.getDesktop()
            val dirToOpen = File(ncHome)
            desktop.open(dirToOpen)
        } catch (e: Exception) {
            val message = e.message
            Messages.showInfoMessage(message, "Error")
        }
    }
}
